sparse/densification.py

```diff
@@ -0,0 +1,140 @@
+import sys
+from numbers import Integral, Number
+from collections import Iterable
+
+import numpy as np
+
+from .utils import _get_broadcast_shape, SparseArray
+
+try:  # Windows compatibility
+    int = long
+except NameError:
+    pass
+
+try:
+    _max_size = sys.maxsize
+except NameError:
+    _max_size = sys.maxint
+
+
+class DensificationConfig(object):
+    def __init__(self, densify=None, max_size=None, min_density=None):
+        if isinstance(densify, (Iterable, DensificationConfig)):
```

Comment:
I don't like how you abuse the `densify` parameter here. If you want a copy constructor, we should have one specifically for it.